CRM/Financial/Form/BatchTransaction.php

```diff
@@ -52,7 +52,17 @@ public function preProcess() {
     $this->assign('entityID', self::$_entityID);
     if (isset(self::$_entityID)) {
       $this->_batchStatusId = CRM_Core_DAO::getFieldValue('CRM_Batch_BAO_Batch', self::$_entityID, 'status_id');
+      $this->batchStatuses = CRM_Core_OptionGroup::values('batch_status', TRUE);
       $this->assign('statusID', $this->_batchStatusId);
+      $validStatuses = array(
+        $this->batchStatuses['Open'],
+        $this->batchStatuses['Reopened'],
+      );
+      $this->assign('batchStatuses', $validStatuses);
+      $this->assign('validStatus', FALSE);
+      if (in_array($this->_batchStatusId, array($this->batchStatuses['Open'], $this->batchStatuses['Reopened']))) {
```

Comment:
best idea would be to check
if (in_array($this->_batchStatus, array('Open', 'Reopened'))) {